src/models/cost_functions/variable_cost.jl

```diff
@@ -0,0 +1,84 @@
+abstract type ProductionVariableCost{T <: ValueCurve} end
+
+IS.serialize(val::ProductionVariableCost) = IS.serialize_struct(val)
+IS.deserialize(T::Type{<:ProductionVariableCost}, val::Dict) = IS.deserialize_struct(T, val)
+
+"Get the underlying `ValueCurve` representation of this `ProductionVariableCost`"
+get_value_curve(cost::ProductionVariableCost) = cost.value_curve
+"Get the units for the x-axis of the curve"
+get_power_units(cost::ProductionVariableCost) = cost.power_units
+"Get the `FunctionData` representation of this `ProductionVariableCost`'s `ValueCurve`"
+get_function_data(cost::ProductionVariableCost) = get_function_data(get_value_curve(cost))
+"Get the `initial_input` field of this `ProductionVariableCost`'s `ValueCurve` (not defined for input-output data)"
+get_initial_input(cost::ProductionVariableCost) = get_initial_input(get_value_curve(cost))
+"Calculate the convexity of the underlying data"
+is_convex(cost::ProductionVariableCost) = is_convex(get_value_curve(cost))
+
+"""
+Direct representation of the variable operation cost of a power plant in currency. Composed
+of a [`ValueCurve`][@ref] that may represent input-output, incremental, or average rate
+data. The default units for the x-axis are megawatts and can be specified with
+`power_units`.
+"""
+@kwdef struct CostCurve{T <: ValueCurve} <: ProductionVariableCost{T}
+    "The underlying `ValueCurve` representation of this `ProductionVariableCost`"
+    value_curve::T
+    "The units for the x-axis of the curve; defaults to natural units (megawatts)"
+    power_units::UnitSystem = UnitSystem.NATURAL_UNITS
+end
+
+CostCurve(value_curve) = CostCurve(; value_curve)
```

Comment:
Is it painful to implement all of these positional argument constructors? You could require that all cost structs be constructed with kwargs. It might help with consistency.